test/remote/gateways/remote_adyen_test.rb

```diff
@@ -0,0 +1,220 @@
+require 'test_helper'
+
+class RemoteAdyenTest < Test::Unit::TestCase
+  def setup
+    @gateway = AdyenGateway.new(fixtures(:adyen))
+
+    @amount = 100
+
+    @credit_card = credit_card('4111111111111111',
+      :month => 8,
+      :year => 2018,
+      :first_name => 'John',
+      :last_name => 'Smith',
+      :verification_value => '737',
+      :brand => 'visa'
+    )
+
+    @declined_card = credit_card('4000300011112220')
+
+    @options = {
+      reference: '345123',
+      shopperEmail: "john.smith@test.com",
+      shopperIP: "77.110.174.153",
```

Comment:
We should probably underscore-ize these options. Which actually brings up another point- we have a couple gateway options [that are standardized](https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateway.rb#L25), `email` and `ip` being two of them.

Looks like the gateway adapter is just mapping option keys to API request fields which works but does break that soft contract we have with the `options` hash. Other gateways generally take something like `email` from the options and map them to a appropriate API request key. For example, [Be2Bill](https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/be2bill.rb#L63)

```ruby
post[:CLIENTEMAIL] = options[:email]
```

Formalizing the options hash is probably something we revisit.